### ISSUE TITLE
local version of .info and .SlackBuild delete at startup when using SBo-git repo

### ISSUE THREAD

**Issue by ArTourter** [sbopkg/sbopkg#66]
all .sbopkg files are delete when starting sbopkg (latest git commit) when using SBo-git repo.

to reproduce:
```
REPO_NAME=SBO-git
REPO_BRANCH=current
```

after initial pull, create an .sbopkg file in any folder, for instant
```
touch /var/lib/sbopkg/SBo-git/academic/R/R.info.sbopkg
```

start sbopkg
quit sbopkg

```
ls /var/lib/sbopkg/SBo-git/academic/R/R.info.sbopkg
/bin/ls: cannot access '/var/lib/sbopkg/SBo-git/academic/R/R.info.sbopkg': No such file or directory
```

I look in the code but couldn't find the bit that as doing it. It only appears to do this when using the SBo-git repo. All others are find, including the 14.2 git one.

**Comment by willysr** [comment]
when you start sbopkg, what action did you do?

**Comment by ArTourter** [comment]
nothing special. simply starting sbopkg with the SBo-git repo clears the .sbopkg files.

**Comment by willysr** [comment]
is this on 0.38.1 ?
or you are using this one? https://sbopkg.org/test/sbopkg-0.38.2-noarch-1_wsr.tgz

**Comment by ArTourter** [comment]
you are correct. it is indeed the checkout_git_branch function that does it, and more specifically https://github.com/sbopkg/sbopkg/blob/64ffa57daba24a553d8b95df63a95af4093f0822/src/usr/sbin/sbopkg#L1046.

I don't understand why though, since if I understand the code right, we are ignoring */*/*.sbopkg. but it doesn't work for the SBo-git repo (yet it works for the SBo/master repo. I am not versed enough with git to figure out a fix though, but I will keep trying.

**Comment by ArTourter** [comment]
ok so the reason it works fine for the SBo/master repo is because the branch ($NEW_BRANCH) is called "master" whereas for SBo-git, the branch is "current".

so  with SBo/master, we exit the function at line sbopkg#L1037 since NEW_BRANCH==CURRENT_BRANCH. whereas with SBo-git, CURRENT_BRANCH is still master and NEW_BRANCH is current so it carries on.

I am still not quite sure why the `git branch` command reports that the branch is called "master" when it should be called current.

**Comment by willysr** [comment]
master refers to the same master branch in SBo, whereas current is ponce's own branch which is a complete merge from all his personal branches that contains changes spesific for -current.

**Comment by ArTourter** [comment]
I know. but if I clone Ponce's repo with `git clone https://github.com/Ponce/slackbuilds.git`, calling `git branch` in the tree returns "* current". In the tree of the same repo from sbopkg, the same `git branch` command returns "* master".

**Comment by willysr** [comment]
that's because he set the default branch is current instead of master

**Comment by ArTourter** [comment]
ok so how do we make sbopkg reflect this, so that when sbopkg gets the repo the local branch name is the same as on the remote.

**Comment by willysr** [comment]
this is out of scope for sbopkg as it just pulls from the default branch in the repository

**Comment by ArTourter** [comment]
Actually, with the latest changes to the initial clone of SBo-git from c353ac7 but **only after removing the original tree** and let it clone the whole thing again.
So I think with that commit I would consider the problem solved.
